### PR TITLE
Migrate TF Smooth Quant to 3.x API

### PR DESCRIPTION
**Pull request by zehao-intel**
## Type of Change

Feature

## Description

This is a separate PR which only includes TF SQ.
The Static Quant part will be in another PR with auto tune.

## How has this PR been tested?

UT

## Dependency Change?

No

### CODE REVIEW

File: neural_compressor/tensorflow/quantization/tf_utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2024 Intel Corporation
```
Comment (yiliu30) [inline neural_compressor/tensorflow/quantization/tf_utils/__init__.py:1]: As it is already under `tensorflow` folder, please remove `tf_` or `tensorflow_` prefix.

File: neural_compressor/tensorflow/utils/tensorflow_models.py [hunk 2]
```diff
@@ -0,0 +1,1589 @@
+#!/usr/bin/env python
```
Comment (yiliu30) [inline neural_compressor/tensorflow/utils/tensorflow_models.py:1]: Please remove `tensorflow` prefix.

File: neural_compressor/tensorflow/algorithms/smooth_quant/__init__.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (yiliu30) [inline neural_compressor/tensorflow/algorithms/smooth_quant/__init__.py:2]: Are these files only made smooth? If so, we can rename it as follows:
```bash
core.py
calibrator.py
scaler.py
```

**Review by yiliu30 — COMMENTED** [review]
Left some comments, let's discuss the details offline.

File: neural_compressor/tensorflow/quantization/algorithm_entry.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        calib_dataloader: a data loader for calibration.
+        calib_iteration: the iteration of calibration.
+
+    Returns:
+        q_model: the quantized model.
+    """
+    keras_adaptor = KerasAdaptor(framework_specific_info)
+    keras_adaptor.query_fw_capability(model)
+    config_parser = ParseKerasConfig(quant_config, calib_iteration)
+    tune_cfg = config_parser.parse_to_tune_cfg()
```
[the remark this answers is not in the document]
Comment (zehao-intel) [reply neural_compressor/tensorflow/quantization/algorithm_entry.py:48]: Will refine this in the PR of TF Static Quant.

**Comment by zehao-intel** [comment]
All issues in the conversation has been solved.

**Review by yiliu30 — APPROVED** [review]

**Pull request merged**